### ISSUE TITLE
change queuing defaults

### ISSUE THREAD

**Issue by colinsurprenant** [elastic/logstash#8707]
Per #8702 I suggest we change the following defaults:

```
pipeline.batch.delay: 5
queue.page_capacity: 250mb
```

to these new values:

```
pipeline.batch.delay: 50
queue.page_capacity: 25mb
```

- increasing the `pipeline.batch.delay` does not have any negative incidence and provide more time to form full batches.
- decreasing `queue.page_capacity` mitigate a performance problem which seems related to the number of written elements in each PQ page. The only potential problem I see is that it might cause problems with documents/message sizes > 25mb but seems not likely for the majority of users who will benefit from a smaller default page size.

**Comment by jakelandis** [comment]
> pipeline.batch.delay: 50

+1

> decreasing `queue.page_capacity`

Wouldn't this add additional strain on the vmem system, and would compound [this](https://github.com/elastic/logstash/issues/8688) issue ?  If we reduce by 10x then the OS and us has 10x more things to keep track, which under load may (or maynot) be significant.  

Do we know the performance slows down for large pages ?

**Comment by original-brownbear** [comment]
@jakelandis @colinsurprenant the effect of large pages is very OS and system dependant.

On OSX you're hit the hardest as far as I can tell, on Linux not so hard. Ideally we'd want to unmap the pages not used anyways, so if the number of mapped files becomes an issue it's a bug I guess.

Technically smaller pages means more overhead for files, but cheaper fsync (this is almost linear in the size of the page over some ranges of pagesize).

I'm completely +1 on upping the batch delay. On the `25MB` I'm a bit torn, I def agree that we should go for something smaller than what we currently have, `25MB` seems fairly low though. In my testing I never saw much gain below 64MB but more of a hit from writing checkpoints and such (this is also what InnoDB uses as segment file size afaik).
I'm for 64MB on this one just by looking at what others are doing and what my ext4 tests did (unless 25MB gives us an actual gain on some systems, especially for smaller queue sizes it shouldn't matter).

**Comment by colinsurprenant** [comment]
@jakelandis we don't know yet exactly for sure what the cause is for the large pages slowdowns. I will spend more time digging into it and profiling but we've been observing that and also the fact that smaller pages seems to perform better. It is also very clear performance-wise in #8702.
Also note that the *only* impact on vmem is faster page churn which does not seem to be a problem per #8702 again but I haven't specifically measured it outside the resulting pipeline performance.

**Comment by colinsurprenant** [comment]
@original-brownbear 

> the effect of large pages is very OS and system dependant. 

True and granted I only did my tests on OSX - I will do them on Linux shorty.

> I'm for 64MB on this one just by looking at what others are doing and what my ext4 tests did (unless 25MB gives us an actual gain on some systems, especially for smaller queue sizes it shouldn't matter).

+1

**Comment by colinsurprenant** [comment]
Per #8688 it seems we should look at removing `-XX:+DisableExplicitGC`.

**Comment by original-brownbear** [comment]
Proper link https://github.com/elastic/logstash/issues/8688#issuecomment-346472363 :D but yea, at least we could actually just document this as a fix if you run into trouble.
@dedemorton I think you have some doc changes for the PQ incoming anyways, maybe we can just add the recommendation to remove the line `-XX:+DisableExplicitGC` from `jvm.options` for users of large (queue_size > physical memory / 2) PQs whenever that work lands?

**Comment by colinsurprenant** [comment]
`-XX:+DisableExplicitGC` has been removed in in elastic/logstash/pull/8776

**Comment by colinsurprenant** [comment]
PR in #8797

**Comment by colinsurprenant** [comment]
Closing per #8797, will be included in 6.2 release.
